sabnzbd/filesystem.py

```diff
@@ -59,18 +59,40 @@ def get_ext(filename: str) -> str:
         return ""
 
 
+def ext_is_listed(ext: str, ext_list: list) -> bool:
+    """Check if the extension is listed. In case of a regexp the entire extension must be matched;
+    partial matches aren't accepted (e.g. 'r[0-9]{2}' will be treated the same as '^r[0-9]{2}$' and
+    thus return false for extentions such as 'r007' despite the substring match on 'r00').
+
+    Since the extension list is stored as a comma-separated list in the config, regular expressions
+    cannot use a comma or they'll be split and interpreted as multiple listed extensions.
+    """
+    for item in ext_list:
+        if item.startswith("re:"):
+            try:
+                # Case-insensitive match, extensions from both cfg and get_ext are always lowercase
+                if len(re.compile(item[3:], re.I).match(ext).group()) == len(ext):
```

Comment:
Maybe it's best to use the same logic as `convert_filter` from `rss.py`? This also allows simple regex like `.r*`?